pyomo/contrib/gdpopt/GDPopt.py

```diff
@@ -64,6 +64,12 @@ class GDPoptSolver(object):
         default=30, domain=NonNegativeInt,
         description="Iteration limit."
     ))
+    CONFIG.declare("time_limit", ConfigValue(
+        default=600,
+        domain=PositiveInt,
+        description="Time limit (seconds)",
+        doc="Seconds allowed until terminated"
```

Comment:
We should note that this time limit is "loose". That is, if we get stuck solving a subproblem for a long time, we can exceed the time limit.

Reply:
Fixed in b3d04a30011ee833153708cf87cee68f09a96b73